### PR TITLE
Add referral webhook to docs

### PR DESCRIPTION
**Pull request by hebime**
First pass of the docs. Let me know what you guys think.

### CODE REVIEW

File: slate/source/includes/rest/core/webhooks/_conversation_events.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ The payload for when a user reads a conversation.
 | **source** | A nested object representing the source of the event. See the [source schema](#sourcedestination-schema) below for details. |
 | **appUser** | A nested object representing the [appUser](#app-user-schema) (for "conversation:read" events, a [truncated appUser](#truncated-app-user-schema) is provided). |
 | **timestamp** | A unix timestamp given in seconds, describing when Smooch received the event. |
+| **referral** | If a user was referred to starting a conversation by scanning a [Messenger code](https://developers.facebook.com/docs/messenger-platform/messenger-code) or clicking on a [Facebook ad](https://developers.facebook.com/docs/messenger-platform/guides/ads), the "conversation:start" payload will also include the referral information in a referral object. See [below](#conversation-referral-payload-schema) for the details of this object.
```
Comment (jpjoyal) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: "If a user was referred to starting a conversation..." -> "If a user started a conversation...."

Comment (hebime) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: Rofl I mixed two sentences here :man_facepalming:

File: slate/source/includes/rest/core/webhooks/_conversation_events.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ The payload for when a user reads a conversation.
 | **source** | A nested object representing the source of the event. See the [source schema](#sourcedestination-schema) below for details. |
 | **appUser** | A nested object representing the [appUser](#app-user-schema) (for "conversation:read" events, a [truncated appUser](#truncated-app-user-schema) is provided). |
 | **timestamp** | A unix timestamp given in seconds, describing when Smooch received the event. |
+| **referral** | If a user started a conversation by scanning a [Messenger code](https://developers.facebook.com/docs/messenger-platform/messenger-code) or clicking on a [Facebook ad](https://developers.facebook.com/docs/messenger-platform/guides/ads), the "conversation:start" payload will also include the referral information in a referral object. See [below](#conversation-referral-payload-schema) for the details of this object.
```
Comment (jpjoyal) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: Suggest : Referral information. Available for Facebook Messenger only. See [referral schema](#conversation-referral-payload-schema) for details.

File: slate/source/includes/rest/core/webhooks/_conversation_events.md [hunk 3]
```diff
@@ -64,3 +65,45 @@ The payload for when a user reads a conversation.
...
+```
+
+#### Conversation referral payload schema
+|    Field     |        Description        |
+|--------------|---------------------------|
+| **trigger**  | `"conversation:referral"` |
+| **app**      | A nested object representing the Smooch app associated with the event. See the [truncated app schema](#truncated-app-schema) below for details. |
+| **source**   | A nested object representing the source of the event. See the [source schema](#sourcedestination-schema) below for details. |
+| **appUser**  | A [truncated appUser](#truncated-app-user-schema) schema object |
+| **referral** | A nested object containing the referral code from an [ad](https://developers.facebook.com/docs/messenger-platform/guides/ads) or [Messenger code](https://developers.facebook.com/docs/messenger-platform/messenger-code), and details containing the source and type of referral.
```
Comment (jpjoyal) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: Should add a schema object for this.

File: slate/source/includes/rest/core/webhooks/_conversation_events.md [hunk 4]
```diff
@@ -64,3 +65,45 @@ The payload for when a user reads a conversation.
...
+
+The payload for when a user is referred to an *existing* conversation. Currently, Messenger is the only channel that supports this.
```
Comment (jpjoyal) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: Its not so much the payload that is for existing conversation as the trigger itself.

File: slate/source/includes/rest/core/_conversations.md [hunk 5]
```diff
@@ -1307,3 +1307,16 @@ This table represents the fields you can expect to receive nested inside postbac
...
+
+Data representing a referral object when a user is referred to a conversation via a [Messenger code](https://developers.facebook.com/docs/messenger-platform/messenger-code) or clicking a [conversion ad](https://developers.facebook.com/docs/messenger-platform/guides/ads) on Facebook.
+
+| Field       | Description                                                                  |
+|-------------|------------------------------------------------------------------------------|
+| **code**    | The referral's identifier                                                    |
+| **details** | Nested object containing `source`, `type` and `adId`                         |
+| **source**  | The source of the referral. Ex: `MESSENGER_CODE`, `ADS` etc...               |
+| **type**    | The type of referral, usually `OPEN-THREAD`                                  |
+| **adID**    | If the referral came from an ad, this field will be present with the ad's Id |
```
Comment (mspensieri) [inline slate/source/includes/rest/core/_conversations.md]: `adID` or `adId` ?

Comment (hebime) [inline slate/source/includes/rest/core/_conversations.md]: arrgh! typo. thanks.

File: slate/source/includes/rest/core/webhooks/_conversation_events.md [hunk 6]
```diff
@@ -31,6 +31,7 @@ The payload for when a user reads a conversation.
 | **source** | A nested object representing the source of the event. See the [source schema](#sourcedestination-schema) below for details. |
 | **appUser** | A nested object representing the [appUser](#app-user-schema) (for "conversation:read" events, a [truncated appUser](#truncated-app-user-schema) is provided). |
 | **timestamp** | A unix timestamp given in seconds, describing when Smooch received the event. |
+| **referral** | Referral information. Available for Facebook Messenger only. See [referral schema](#referral-schema) for details.
```
Comment (jpjoyal) [inline slate/source/includes/rest/core/webhooks/_conversation_events.md]: I would mention its only available for conversation start too.

**Pull request merged**